app/policies/project_policy.rb

```diff
@@ -0,0 +1,42 @@
+class ProjectPolicy < ApplicationPolicy
+  authorize :user, allow_nil: true
+
+  def index?
+    true
+  end
+
+  def new?
+    true
+  end
+
+  def create?
+    user.present?
+  end
+
+  def destroy?
+    owner?
```

Comment:
You need to check that user present like in `create?` 
```suggestion
    user.present? && owner?
```
Do same in `update?`, but not forget about correspond role to update action